ignite/contrib/handlers/param_scheduler.py

```diff
@@ -549,22 +559,40 @@ def simulate_values(cls, num_events, schedulers, durations, param_names=None, **
         """
         if param_names is not None and not isinstance(param_names, (list, tuple)):
             raise ValueError("Argument param_names should be list or tuple of strings")
-        output = []
-
-        # Need to copy all schedulers otherwise unsafe
-        copy_schedulers = [_replicate_scheduler(s) for s in schedulers]
 
-        scheduler = cls(copy_schedulers, durations, save_history=False)
-        if param_names is None:
-            param_names = [scheduler.param_name]
-        for i in range(num_events):
-            scheduler(engine=None)
-            values = [i]
-            for param_name in param_names:
-                params = [p[param_name] for p in scheduler.optimizer_param_groups]
-                values = values + params
-            output.append(values)
-        return output
+        # This scheduler uses `ParamScheduler` which
+        # should be replicated in order to simulate LR values and
+        # not perturb original scheduler.
+        with tempfile.TemporaryDirectory() as tmpdirname:
```

Comment:
@sdesrozis this part of code : create temp dir, save current state, run something and restore state can be factorized into a context manager. But I do not insist...